src/preproc/preprocessing_pass.h

```diff
@@ -0,0 +1,100 @@
+/*********************                                                        */
+/*! \file preprocessing_pass.h
+ ** \verbatim
+ ** Top contributors (to current version):
+ **  Justin Xu
+ ** This file is part of the CVC4 project.
+ ** Copyright (c) 2009-2017 by the authors listed in the file AUTHORS
+ ** in the top-level source directory) and their institutional affiliations.
+ ** All rights reserved.  See the file COPYING in the top-level source
+ ** directory for licensing information.\endverbatim
+ **
+ ** \brief Implementation for preprocessing pass super class
+ **
+ ** Implementation for preprocessing pass super class. Assertion Pipeline has
+ *been moved from SmtEngine to here, and methods like dump assertions, among
```

Comment:
The comment should not talk about changes made since this is going to be irrelevant in the future. Comments should only talk about the status quo (except giving some historic context on why things are the way they are in some special circumstances). Changes made should be described in the commit message.

This comment should instead describe what the class provides (timing, dumping, common interface etc.) and what a user of the class would have to implement to add a new preprocessing pass. This is going to be the starting point for anyone who wants to implement a new pass in the future, so it is quite important.